lib/LedgerSMB/Scripts/setup.pm

```diff
@@ -775,27 +775,54 @@ verify_check => md5_hex($check->test_query),
     $sth->finish();
 
     my $heading = { map { $_ => $_ } @{$check->display_cols} };
-    my $buttons = [
+    my %buttons = map { $_ => 1 } @{$check->buttons};
+    my $buttons;
+    push @$buttons,
            { type => 'submit',
              name => 'action',
-            value => $check->columns ? 'fix_tests' : 'cancel',
-             text => $request->{_locale}->text($check->columns
-                                                ? 'Save and Retry'
-                                                : 'Cancel'),
+            value => 'fix_tests',
+          tooltip => { id => 'action-fix-tests',
+                       msg => $request->{_locale}->text($check->{tooltips}{'Save and Retry'}),
+                       position => 'above'
+                     },
+             text => $request->{_locale}->text('Save and Retry'),
             class => 'submit' },
-    ];
+        if $check->columns;
+    push @$buttons,
+           { type => 'submit',
+             name => 'action',
+            value => 'cancel',
+          tooltip => { id => 'action-cancel',
+                       msg => $request->{_locale}->text($check->{tooltips}{'Cancel'}),
+                       position => 'above'
+                     },
+             text => $request->{_locale}->text('Cancel'),
+            class => 'submit' }
+    if $buttons{Cancel} or scalar($check->columns) == 0;
```

Comment:
The indenting here is off (should have been the same as `if $check->columns;` above).